src/encoding.ts

```diff
@@ -0,0 +1,379 @@
+const BASE64_KEYSTR = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+
+function normalizeEncodingArgs(data: any, alternateCodec: any): [ string, ByteBuffer ]{
+	data = String(data);
+
+	return [ data, <ByteBuffer> {
+		toString: function (codec?: any): string {
+			return (codec || alternateCodec).decode(this);
+		},
+		toJSON: function (): string {
+			return JSON.stringify(Array.prototype.slice.call(this));
+		},
+		length: data.length
+	}];
+}
+
+function validateDecodingArgs(data: any) {
+	if (data == null) {
+		return false;
+	}
+
+	if (data.length == null) {
+		throw new TypeError('Argument must have a length property');
```

Comment:
If there's no expectation that data would be explicitly null or undefined (only not defined at all), then that would make more sense here.